core/cas-server-core-events/src/test/java/org/apereo/cas/support/events/CasAuthenticationEventListenerTests.java

```diff
@@ -112,43 +129,137 @@ public void verifyCasAuthenticationTransactionFailureEvent() {
+        assertRepositoryIsEmpty();
         val event = new CasAuthenticationPolicyFailureEvent(this,
-            CollectionUtils.wrap("error", new FailedLoginException()),
-            new DefaultAuthenticationTransaction(CoreAuthenticationTestUtils.getService(),
-                CollectionUtils.wrap(CoreAuthenticationTestUtils.getCredentialsWithSameUsernameAndPassword())),
-            CoreAuthenticationTestUtils.getAuthentication());
-        applicationContext.publishEvent(event);
+                CollectionUtils.wrap("error", new FailedLoginException()),
+                new DefaultAuthenticationTransaction(CoreAuthenticationTestUtils.getService(),
+                        CollectionUtils.wrap(CoreAuthenticationTestUtils.getCredentialsWithSameUsernameAndPassword())),
+                CoreAuthenticationTestUtils.getAuthentication(), null);
+        publishEventAndWaitToProcess(event);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
 
     @Test
     public void verifyCasRiskyAuthenticationDetectedEvent() {
+        assertRepositoryIsEmpty();
         val event = new CasRiskyAuthenticationDetectedEvent(this,
-            CoreAuthenticationTestUtils.getAuthentication(),
-            CoreAuthenticationTestUtils.getRegisteredService(),
-            new Object());
-        applicationContext.publishEvent(event);
+                CoreAuthenticationTestUtils.getAuthentication(),
+                CoreAuthenticationTestUtils.getRegisteredService(),
+                new Object(), null);
+        publishEventAndWaitToProcess(event);
         assertFalse(casEventRepository.load().findAny().isEmpty());
     }
 
     @Test
     public void verifyCasTicketGrantingTicketDestroyed() {
+        assertRepositoryIsEmpty();
         val event = new CasTicketGrantingTicketDestroyedEvent(this,
-            new MockTicketGrantingTicket("casuser"));
-        applicationContext.publishEvent(event);
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
+        assertFalse(casEventRepository.load().findAny().isEmpty());
+    }
+
+
+    @Test
+    public void verifyEventRepositoryHasOneEventOnly() {
+        assertRepositoryIsEmpty();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
+        assertEquals(1, casEventRepository.load().count());
+    }
+
+    @Test
+    public void verifyCasTicketGrantingTicketDestroyedHasClientInfo() {
+        assertRepositoryIsEmpty();
+        val event = new CasTicketGrantingTicketDestroyedEvent(this,
+                new MockTicketGrantingTicket("casuser"), ClientInfoHolder.getClientInfo());
+        publishEventAndWaitToProcess(event);
         assertFalse(casEventRepository.load().findAny().isEmpty());
```

Comment:
The loaded results should be extracted into a separate variable, and the used below.

Reply:
It should now be loaded in a separate variable.